src/config/loading.rs

```diff
@@ -22,8 +22,8 @@ lazy_static! {
 
 /// Merge the paths coming from different cli flags with different formats into
 /// a unified list of paths with formats.
-pub fn merge_path_lists<'a>(
-    path_lists: Vec<(&'a [PathBuf], FormatHint)>,
+pub fn merge_path_lists(
+    path_lists: Vec<(&'_ [PathBuf], FormatHint)>,
```

Comment:
Is the explicitly elided lifetime necessary? I assume `&[PathBuf]` would be sufficient instead of `&'_ [PathBuf]`, unless I'm missing something.

The clippy rule https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes says

> Lifetime bounds such as `impl Foo + 'a` and `T: 'a` must be elided with the placeholder notation `'_` because the fully elided notation leaves the type bound to `'static`.

which doesn't seem to be the case here.

Reply:
They are not required, thanks! Removed.

Reply:
Is this a clippy rule we should enable?

Reply:
This rule (`needless_lifetimes`) already enabled (`warn` level) but do not report about `&'_ x` because we do not define lifetime explicitly (like `<'a>`)